app/models/pubmed_source_record.rb

```diff
@@ -29,12 +29,14 @@ def self.get_pubmed_hash_for_pmid(pmid)
   def self.get_pubmed_source_record_for_pmid(pmid)
     find_by(pmid: pmid) || get_pubmed_record_from_pubmed(pmid)
   end
+  private_class_method :get_pubmed_source_record_for_pmid
```

Comment:
Why is this _not_ a useful public method?  Exposing a utility that returns an existing source record or fetches it from an API seems like a useful public API.  In this way, the source record adopts some responsibility for managing all the source record data, while exposing a public API to that source data.  This could be used by the WOS harvester, for example.  The other methods that are called by this method could be suitable candidates for private methods.

Reply:
```
$ git grep 'get_pubmed_source_record_for_pmid' # on master
app/models/pubmed_source_record.rb:    pubmed_source_record = get_pubmed_source_record_for_pmid(pmid)
app/models/pubmed_source_record.rb:  def self.get_pubmed_source_record_for_pmid(pmid)
```